app/src/main/java/org/schabi/newpipe/util/ShareUtils.java

```diff
@@ -21,22 +22,80 @@ private ShareUtils() {
      * Open the url with the system default browser.
      * <p>
      * If no browser is set as default, fallbacks to
-     * {@link ShareUtils#openInDefaultApp(Context, String)}
+     * {@link ShareUtils#openInDefaultApp(Context, Intent)}
+     *
+     * @param context                the context to use
+     * @param url                    the url to browse
+     * @param httpDefaultBrowserTest the boolean to set if the
+     *                               test for the default browser will be for HTTP protocol
+     *                               or for the created intent
+     */
+    public static void openUrlInBrowser(final Context context, final String url,
+                                        final Boolean httpDefaultBrowserTest) {
+        final String defaultPackageName;
+        final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url))
+                .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        if (httpDefaultBrowserTest) {
+            defaultPackageName = getDefaultBrowserPackageName(context);
+        } else {
+            defaultPackageName = getDefaultAppPackageName(context, intent);
+        }
+
+        if (defaultPackageName.equals("android")) {
+            // no browser set as default (doesn't work on some devices)
+            openInDefaultApp(context, intent);
+        } else {
+            try {
+                intent.setPackage(defaultPackageName);
+                context.startActivity(intent);
+            } catch (final ActivityNotFoundException e) {
+                // not a browser but an app chooser because of OEMs changes
+                intent.setPackage(null);
+                openInDefaultApp(context, intent);
+            }
+        }
+    }
+
+    /**
+     * Open the url with the system default browser.
+     * <p>
+     * If no browser is set as default, fallbacks to
+     * {@link ShareUtils#openInDefaultApp(Context, Intent)}
+     * <p>
+     * This call {@link ShareUtils#openUrlInBrowser(Context, String, Boolean)} with true
+     * for the boolean parameter
      *
      * @param context the context to use
      * @param url     the url to browse
-     */
+     **/
     public static void openUrlInBrowser(final Context context, final String url) {
-        final String defaultBrowserPackageName = getDefaultBrowserPackageName(context);
+        openUrlInBrowser(context, url, true);
+    }
+
+    /**
+     * Open a content with the system default browser.
+     * <p>
+     * If no app is set as default, fallbacks to
+     * {@link ShareUtils#openInDefaultApp(Context, Intent)}
+     *
+     * @param context the context to use
+     * @param intent  the intent of the file to open
+     */
+    public static void openContentInApp(final Context context, final Intent intent) {
```

Comment:
Could you give a better name to this? For me the difference between `openContentInApp` and `openInDefaultApp` isn't clear

Reply:
I was thinking first of openFileInApp but I also use this method to play content in an external app.

Reply:
Method name changed to openIntentInApp